### DISCUSSION TITLE
Silencing "Function declaration "{func name}" is obscured by a declaration of the same name" Pylance(reportGeneralTypeIssues)

### DISCUSSION THREAD

**Discussion by theFong** [microsoft/pyright#1464] [Q&A]
Is there a way to silence `Function declaration "{func name}" is obscured by a declaration of the same name" Pylance(reportGeneralTypeIssues)` without having to silence all `reportGeneralTypeIssues`?

**Comment by erictraut** [comment]
You could use a `# type: ignore` comment to silence specific instances.

I'm curious about the circumstance in which you would want to silence this rather than modifying the code to avoid it. If you're able to provide details or examples, I'd be interested.

**Reply by jakebailey** [reply]
I don't know how popular it is, but the most recent instance I've seen of this was: https://github.com/microsoft/pylance-release/issues/834

**Reply by theFong** [reply]
Essentially this. I was experimenting with building a multiple dispatch decorator that would support type hints.

**Reply by erictraut** [reply]
I can think of a couple of ways to handle this:
1. I could suppress this check if the functions contain any decorator. This has the downside in that it would apply to common decorators like `@static_method` and `@property`, so it will lead to potential false negatives.
2. I could create a new diagnostic rule to separate this from other diagnostics associated with `reportGeneralTypeIssues`, say `reportObscuredFunctionDeclaration`. That way, it could be disabled independently from others. The downside of this approach is that you wouldn't ever be able to use multidispatch with "strict" mode because this new diagnostic rule would be enabled by default in strict mode. You could disable this diagnostic mode on a file-by-file basis using `# pyright: reportObscuredFunctionDeclaration=false`.
3. We could leave it as is for now, in which case you'd need to disable all diagnostics for `reportGeneralTypeIssues` within the file.

I'm not a big fan of option 1, so I'd lean toward 2 or 3.

**Reply by theFong** [reply]
I'm also not a fan of option 1.  Intuitively I was thinking I could do option 2 which would be sufficient for my use case.
